src/output.cpp

```diff
@@ -869,118 +879,122 @@ std::string string_input_win( WINDOW *w, std::string input, int max_length, int
             wrefresh( w );
         }
         bool return_key = false;
-        const std::string action = ctxt.handle_input();
+        action = ctxt.handle_input();
         const input_event ev = ctxt.get_raw_input();
         ch = ev.type == CATA_INPUT_KEYBOARD ? ev.get_first_input() : 0;
-        if( callbacks[ch] ) {
-            callbacks[ch]();
-        }
-        if( ch == KEY_ESCAPE ) {
-            return "";
-        } else if( ch == '\n' ) {
-            return_key = true;
-        } else if( ch == KEY_UP ) {
-            if( !identifier.empty() ) {
-                std::vector<std::string> &hist = uistate.gethistory( identifier );
-                uimenu hmenu;
-                hmenu.title = _( "d: delete history" );
-                hmenu.return_invalid = true;
-                for( size_t h = 0; h < hist.size(); h++ ) {
-                    hmenu.addentry( h, true, -2, hist[h].c_str() );
+
+        if ( ch_code_blacklist.find(ch) == ch_code_blacklist.end() ) {
```

Comment:
Instead of intending (and thereby changing) all the lines below, you can use `continue` when the condition is *not* fulfilled:
```C++
while( ... ) {
    if( your_inverted_condition ) {
        continue;
    }
    // remaining code is unchanged.
}
```